### PR TITLE
Some reworking and lots of new functionality

### PR DESCRIPTION
**Pull request by anthonyclays**
I wanted to experiment with hexagonal cellular automata in Julia and needed a whole lot of additional capabilities. Over the last few days I added loads of functionality and tried to make the codebase more consistent / ordered.
I also fixed a few bugs I came across - for example there's a typo in the function name for `neighbors`, there are no conversion rules to convert to a `HexagonOffsetEvenR`.
I tried to stay as close as possible to [Amit's reference](http://www.redblobgames.com/grids/hexagons/#rings) - you'll find a lot of the functionality I added comes straight from there ;)

There's still some work to be done though (in particular writing tests!), but I believe the library is already in a far better shape.

### CODE REVIEW

**Comment by anthonyclays** [comment]
Seems like I forgot to update the readme, so some of the examples there are out-of-date - sorry about that!

File: src/Hexagons.jl [hunk 1]
```diff
@@ -19,177 +24,326 @@ immutable HexagonAxial <: Hexagon
...
-
 function convert(::Type{HexagonCubic}, hex::HexagonAxial)
-    return HexagonCubic(hex.q, hex.r, -hex.q - hex.r)
+    HexagonCubic(hex.q, hex.r, -hex.q - hex.r)
 end
 
-
 function convert(::Type{HexagonCubic}, hex::HexagonOffsetOddR)
-    x = hex.q - div(hex.r - (hex.r & 1), 2)
+    x = hex.q - (hex.r >> 1)
```
Comment (dcjones) [inline src/Hexagons.jl:71]: Is this right? It's `col - (row - (row&1)) / 2` it Amit's reference.

Comment (anthonyclays) [inline src/Hexagons.jl:71]: hex.r & is one iff hex.r is odd. When you shift one bit to the right, the least significant bit is dropped anyways, so this comes down to the same thing (and is faster if the compiler doesn't optimize it away)
Edit: example:

``` julia
julia> f(r) = div(r - (r&1), 2)
f (generic function with 1 method)

julia> g(r) = r >> 1
g (generic function with 1 method)

julia> map(i -> (f(i), g(i)), -10:10)
21-element Array{(Int64,Int64),1}:
 (-5,-5)
 (-5,-5)
 (-4,-4)
 (-4,-4)
 (-3,-3)
 (-3,-3)
 (-2,-2)
 (-2,-2)
 (-1,-1)
 (-1,-1)
 (0,0)  
 (0,0)  
 (1,1)  
 (1,1)  
 (2,2)  
 (2,2)  
 (3,3)  
 (3,3)  
 (4,4)  
 (4,4)  
 (5,5)  
```

Comment (dcjones) [inline src/Hexagons.jl:71]: Thanks, that makes sense.

**Comment by dcjones** [comment]
Thanks for all the work!

**Comment by anthonyclays** [comment]
No problem!
Could you tag a new release and update METADATA.jl?

**Comment by dcjones** [comment]
Yep, done.

**Pull request merged**